File: Documentation/5.4/Samples/python/ClientApi/Session/HowTo/GetLastModified.py
from datetime import datetime

from examples_base import ExampleBase


class Foo:
    # region get_last_modified_1
    def get_last_modified_for(self, entity: object) -> datetime: ...

    # endregion


class GetLastModified(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_get_last_modified(self):
        with self.embedded_server.get_document_store("GetLastModified") as store:
            with store.open_session() as session:
                # region get_last_modified_2
                employee = session.load("employees/1-A")
                last_modified = session.advanced.get_last_modified_for(employee)
                # endregion

File: Documentation/5.4/Samples/php/ClientApi/session/HowTo/GetLastModified.php
<?php

namespace RavenDB\Samples\ClientApi\Session\HowTo;

use DateTimeInterface;
use Employee;
use RavenDB\Documents\DocumentStore;

interface FooInterface
{
    # region get_last_modified_1
    public function getLastModifiedFor($instance): ?DateTimeInterface;
    # endregion
}

class GetLastModified
{
    public function example(): void
    {
        $store = new DocumentStore();
        try {
            $session = $store->openSession();
            try {
                # region get_last_modified_2
                $employee = $session->load(Employee::class, "employees/1-A");
                $lastModified = $session->advanced()->getLastModifiedFor($employee);
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
